datastore/statedata.go

```diff
@@ -223,13 +224,21 @@ func (r *RebootRequestedType) Set(n int, t RebootType) error {
 }
 
 type Artifact struct {
+	// Source provides the download URI for the artifact.
 	Source struct {
 		URI    string
 		Expire string
 	}
+	// Artifact compatible devices
 	CompatibleDevices []string `json:"device_types_compatible"`
-	ArtifactName      string   `json:"artifact_name"`
-	PayloadTypes      []string
+	// Alias for Provides.ArtifactName for api/v1 copatability
+	// - This property should be deleted and ignored after making sure
+	//   the Provides.ArtifactName is set after LoadStateData.
```

Comment:
This should probably rather say that when saving and loading, the alias and the provide are ensured to always be the same.